challenges/objects-arrays.js

```diff
@@ -85,41 +118,53 @@ zooAnimals = [{"animal_name":"Jackal, asiatic","population":5,"scientific_name":
 
 /* Request 1: .forEach()
 
-The zoo wants to display both the scientific name and the animal name in front of the habitats.  Return an array with only the animal and scientific names in it.  The individual values in the array should look like this "Name: Jackal, asiatic, Scientific: Canis aureus."
+The zoo wants to display both the scientific name and the animal name in front of the habitats.
+Return an array with only the animal and scientific names in it.
+The individual values in the array should look like this "Name: Jackal, asiatic, Scientific: Canis aureus."
 
 */
 const animalNames = [];
+zooAnimals.forEach(animal => animalNames.push(`Name: ${animal.animal_name}, Scientific: ${animal.scientific_name}`));
 console.log(animalNames);
 
-/* Request 2: .map()    
+/* Request 2: .map()
 
-The zoos need a list of all their animal's names (names only, not scientific) converted to lower case.  Create a new array named lowerCase and map over each name to convert them all to lower case.  Log the resut.
+The zoos need a list of all their animal's names (names only, not scientific) converted to lower case.
+Create a new array named lowerCase and map over each name to convert them all to lower case.
+Log the resut.
 
 */
 
-const lowerCase = [];
-console.log(lowerCase); 
+const lowerCase = zooAnimals.map(animal => animal.animal_name.toLowerCase());
+console.log(lowerCase);
 
-/* Request 3: .filter() 
+/* Request 3: .filter()
 
-The zoos are concenred about animals with a lower population count. Find out which animals have a population less than 5.
+The zoos are concenred about animals with a lower population count.
+Find out which animals have a population less than 5.
 
 */
-const largerPopulation = [];
+const largerPopulation = zooAnimals.filter(animal => animal.population < 5); //return all info of the animal with a pop less than 5
 console.log(largerPopulation);
 
-/* Request 4: .reduce() 
+/* Request 4: .reduce()
 
-The zoos need to know their total animal population across the United States.  Find the total population from all the zoos using the .reduce() method.
+The zoos need to know their total animal population across the United States.
+Find the total population from all the zoos using the .reduce() method.
 
 */
-const populationTotal = 0;
+
+const zooPops = zooAnimals.map(animal => animal.population) //mapped the population into individual array to use in reduce.
+
+const populationTotal = zooPops.reduce((acc, nextNum) => {  //reduced the zooPops to reduce number into one...
+  return acc + nextNum
+}, 0);
+
 console.log(populationTotal);
```

Comment:
This works and is perfectly fine, but here is an easier and more DRY way to do this:

const populationTotal = zooAnimals.reduce((animal, totalPopulation) => {
  return animal + totalPopulation.population;
}, 0);
console.log(populationTotal);